### PR TITLE
Improve build times with azcopy

### PR DESCRIPTION
**Pull request by reidbarber**
Updates CircleCI config to try replacing Azure CLI with AzCopy to enable multithreaded file copying and hopefully cut down build times.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

**Comment by snowystinger** [comment]
Speed boost of 17x on the deploy step??
Will have to check the insights after they refresh tomorrow. But that's crazy
Screenshot 2024-09-13 at 3 13 30 PM
Screenshot 2024-09-13 at 3 14 10 PM

**Comment by snowystinger** [comment]
or 35x if you look at this PR https://app.circleci.com/pipelines/gh/adobe/react-spectrum?branch=bump-lightningcss yikes!

File: .circleci/config.yml [hunk 1]
```diff
@@ -497,11 +502,17 @@ jobs:
     steps:
       - attach_workspace:
           at: /tmp/verdaccio_dist
+      - run:
+          name: Install AzCopy
+          command: |
+            wget -O azcopy.tar.gz https://aka.ms/downloadazcopy-v10-linux
```
Comment (snowystinger) [inline .circleci/config.yml:508]: probably no longer need the docker image we for azure-cli? can probably just use the rsp-large or xlarge executor

**Review by snowystinger — DISMISSED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (reidbarber) [reply .circleci/config.yml:508]: Tried that but we don't have the same permissions on the executor. Will stick with the existing image for now

**Review by devongovett — APPROVED** [review]

**Review by LFDanLu — APPROVED** [review]
🚀🚀🚀🚀🚀

File: .circleci/config.yml (hunk 1 above)
Comment (ktabors) [reply .circleci/config.yml:508]: We can't include this in the docker image setup or a package.json so it's always there?

Comment (reidbarber) [reply .circleci/config.yml:508]: It's a super quick download so I think its fine. I don't see any official images dedicated to this package.

**Pull request merged**